java/src/test/java/ai/rapids/cudf/ColumnVectorTest.java

```diff
@@ -1144,6 +1147,53 @@ void testCast() {
     }
   }
 
+  @Test
+  void testStringCast() {
+    Integer[] integerArray = {1, 2, 3, 7, 8};
+
+    Short[] shortValues = Arrays.stream(integerArray).map(i -> i.shortValue()).collect(Collectors.toList()).toArray(new Short[integerArray.length]);
+    String[] stringShortValues = Arrays.stream(shortValues).map(i -> String.valueOf(i)).collect(Collectors.toList()).toArray(new String[shortValues.length]);
+
+    testCastFixedWidthToStringsAndBack(DType.INT16, () -> ColumnVector.fromBoxedShorts(shortValues), () -> ColumnVector.fromStrings(stringShortValues));
+
+    String[] stringIntValues = Arrays.asList(integerArray).stream().map(i -> String.valueOf(i)).collect(Collectors.toList()).toArray(new String[integerArray.length]);
+
+    testCastFixedWidthToStringsAndBack(DType.INT32, () -> ColumnVector.fromBoxedInts(integerArray), () -> ColumnVector.fromStrings(stringIntValues));
+
+    Long[] longValues = Arrays.stream(integerArray).map(i -> i.longValue()).collect(Collectors.toList()).toArray(new Long[integerArray.length]);
+    String[] stringLongValues = Arrays.stream(longValues).map(i -> String.valueOf(i)).collect(Collectors.toList()).toArray(new String[longValues.length]);
+
+    testCastFixedWidthToStringsAndBack(DType.INT64, () -> ColumnVector.fromBoxedLongs(longValues), () -> ColumnVector.fromStrings(stringLongValues));
+
+    Float[] floatValues = Arrays.stream(integerArray).map(i -> i.floatValue()).collect(Collectors.toList()).toArray(new Float[integerArray.length]);
+    String[] stringFloatValues = Arrays.stream(floatValues).map(i -> String.valueOf(i)).collect(Collectors.toList()).toArray(new String[floatValues.length]);
+
+    testCastFixedWidthToStringsAndBack(DType.FLOAT32, () -> ColumnVector.fromBoxedFloats(floatValues), () -> ColumnVector.fromStrings(stringFloatValues));
+
+    Double[] doubleValues = Arrays.stream(integerArray).map(i -> i.doubleValue()).collect(Collectors.toList()).toArray(new Double[integerArray.length]);
```

Comment:
We need to test/document what happens for nulls, NaN, Infinity and -Infinity.

Reply:
These tests are not very thorough. The goal was to test the plumbing and not the functionality. I can change the data to test the cases you have listed if that's still desired